docs/dev_guide/software-required.md

```diff
@@ -0,0 +1,55 @@
+# Get the required software for Linux and MacOS
```

Comment:
I think we should abbreviate and inline this to the dev guide README.md, people are apathetic to click links so let's not make them hop through > 2 links from the main readme to get to a quick guide on contributing